### PR TITLE
Add schema for the clinic database

### PR DESCRIPTION
**Pull request by gids-dadi**
This branch contains:
The schema used to create the clinic database is based on the diagram in the project requirement.
Foreign keys have also been added accordingly.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    date_of_birth DATE,
+    PRIMARY KEY(id)
+);
+
+-- create medical_histories table
+CREATE TABLE medical_histories (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP,
+    status VARCHAR(50),
+    CONSTRAINT fk_patients FOREIGN KEY(patient_id) REFERENCES patients(id),
```
Comment (rloterh) [inline schema_based_on_diagram.sql]: - [ ] An error is yielding in attempt to create `medical_histories` table referencing `patients` table. You can modify _line 17_ by simply creating a foreign key and reference like this, which works:
```sql
 patient_id integer REFERENCES patients(id),
```
**Create medical_histories table error:**
image

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+-- create invoices table
+CREATE TABLE invoices (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    CONSTRAINT fk_medical_history FOREIGN KEY(medical_history_id) REFERENCES medical_histories (id), 
+    PRIMARY KEY (id)
+);
```
Comment (rloterh) [inline schema_based_on_diagram.sql]: - [ ] An error is yielding in attempt to create `invoice` table referencing patients `medical_histories`. You can modify _line 27_ by simply creating a foreign key and reference like this, which works:
```sql
 medical_history_id integer REFERENCES medical_histories (id), 
```

**Create invoice table error:**
image

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+-- create invoice_items table
+CREATE TABLE invoice_items (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    unit_price DECIMAL,
+    quantity INT,
+    total_price DECIMAL,
+    CONSTRAINT fk_invoices FOREIGN KEY(invoice_id) REFERENCES invoices (id),
+    CONSTRAINT fk_treatment FOREIGN KEY(treatment_id) REFERENCES treatments (id)
```
Comment (rloterh) [inline schema_based_on_diagram.sql]: - [ ] There is a  syntax error on line 38. Kindly add a comma (**,**) at the end on the line 38

Comment (rloterh) [inline schema_based_on_diagram.sql]: - [ ] Unable to create table `invoice items` as foreign keys `invoice_id` and `treatment_id` are not properly pointing the appropriate tables. Kindly follow the reviews examples above on the creating the `medical_histories` table to resolve these errors.

**Creating invoice_item error:**
image

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (rloterh) [inline schema_based_on_diagram.sql]: - [ ] Kindly ensure that **all** FOREIGN KEYS have indexes created, e.g 
```sql
CREATE INDEX ON medical_histories (patient_id),
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Gideon & Bruk, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Patient and treatment tables properly created


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mateo951 — APPROVED** [review]
### Hi @gids-dadi, 

Well done making the required changes 🔥
Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**